lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1204,9 +1204,10 @@ def _build_files_manifest_walk(b_collection_path, namespace, name, ignore_patter
     manifest = _make_manifest()
 
     def _walk(b_path, b_top_level_dir):
+        common_prefix = os.path.commonpath([b_top_level_dir, b_path])
+        b_rel_base_dir = b'' if b_path == b_top_level_dir else os.path.relpath(b_path, common_prefix).lstrip(os.path.sep.encode())
```

Comment:
This starts to accumulate inline complexity, which hurts readability. How about we factor it out like this:
```suggestion
    def _discover_relative_base_directory(b_path: bytes, b_top_level_dir: bytes) -> bytes:
        if if b_path == b_top_level_dir:
            return b''

        common_prefix = os.path.commonpath((b_top_level_dir, b_path))
        b_rel_base_dir = os.path.relpath(b_path, common_prefix)

        return b_rel_base_dir.lstrip(os.path.sep.encode())

    def _walk(b_path, b_top_level_dir):
        b_rel_base_dir = _discover_relative_base_directory(b_path, b_top_level_dir)
```

Reply:
Sure thing, I was trying to have the smallest PR possible, Ill push another commit with all the changes

Reply:
Yeah, keeping patches small is generally a good idea. Though, there's a point when it may start having some negative side effects. On my scale, readability and maintainability are slightly more important.

In other/personal projects, I like using the "Jones Complexity" metric like https://pypi.org/project/jones-complexity/#what-is-this — it allows tracking and preventing overly complex inline expressions, which nurtures habits related to writing maintainable code.

Reply:
Marking this thread as resolved.